controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -404,6 +416,40 @@ private boolean isRunningProductionJob(Application application) {
         return application.deploymentJobs().jobStatus().entrySet().stream()
                 .anyMatch(entry -> entry.getKey().isProduction() && entry.getValue().isRunning(jobTimeoutLimit()));
     }
+
+    /**
+     * When upgrading it is ok to trigger the next job even if the previous failed if the previous has earlier succeeded
+     * on the version we are currently upgrading to
+     */
+    private boolean isProductionJobPreviouslySucceedingOnCurrentlyDeployingVersion(JobStatus jobStatus, Change change) {
```

Comment:
Consider something like `productionJobHasSucceededFor(change, jobStatus)`

and javadoc:
"Returns true if a production job has already succeeded for the given change"

Naming things is difficult :-)

Reply:
That works, thanks!